src/main.go.test

```diff
@@ -0,0 +1,54 @@
+// -*- Mode: Go -*-
+/*******************************************************************************
+*
+* Copyright 2017 Luke Shumaker <lukeshu@parabola.nu>
+*
+* This file is part of Holo.
+*
+* Holo is free software: you can redistribute it and/or modify it under the
+* terms of the GNU General Public License as published by the Free Software
+* Foundation, either version 3 of the License, or (at your option) any later
+* version.
+*
+* Holo is distributed in the hope that it will be useful, but WITHOUT ANY
+* WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+* A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License along with
+* Holo. If not, see <http://www.gnu.org/licenses/>.
+*
+*******************************************************************************/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+var exit int
+
+func TestMain(m *testing.M) {
+	var flags []string
+	if str, ok := os.LookupEnv("HOLO_TEST_FLAGS"); ok {
+		flags = append(flags, strings.Split(str, " ")...)
+	}
+	if dir := os.Getenv("HOLO_TEST_COVERDIR"); dir != "" {
+		coverfile := fmt.Sprintf("%s.%d.%d.cov", filepath.Base(os.Args[0]), os.Getpid(), time.Now().UnixNano())
+		flags = append(flags, "-test.coverprofile", filepath.Join(dir, coverfile))
+	}
+
+	flag.CommandLine.Parse(flags)
+	_ = m.Run()
+	os.Exit(exit)
+}
+
+func TestSystem(t *testing.T) {
+	exit = Main()
+	_ = os.Stdout.Close()
```

Comment:
You had some additional comments about `Main()` and `os.Stdout.Close` in the message for this commit. Can you add these into the code instead?